python/ray/tune/suggest/suggestion.py

```diff
@@ -375,20 +377,24 @@ def on_trial_complete(self,
             return
         elif self.batch:
             self.cached_results[trial_id] = (result, error)
-            if len(self.cached_results) == self.max_concurrent:
+            self.num_unfinished_live_trials -= 1
+            if self.num_unfinished_live_trials <= 0:
                 # Update the underlying searcher once the
                 # full batch is completed.
                 for trial_id, (result, error) in self.cached_results.items():
                     self.searcher.on_trial_complete(
                         trial_id, result=result, error=error)
                     self.live_trials.remove(trial_id)
                 self.cached_results = {}
+                self.num_unfinished_live_trials = 0
+                self.max_concurrent = self.max_concurrent
```

Comment:
what does this self.X = self.X thing do?

Reply:
Leftover from a previous approach. Meant to remove that - thanks!